src/Adapter/Product/CommandHandler/RemoveAllCustomizationFieldsFromProductHandler.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Product\AbstractProductHandler;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Command\DeleteCustomizationFieldCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Command\RemoveAllCustomizationFieldsFromProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\DeleteCustomizationFieldHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\RemoveAllCustomizationFieldsFromProductInterface;
+
+final class RemoveAllCustomizationFieldsFromProductHandler extends AbstractProductHandler implements RemoveAllCustomizationFieldsFromProductInterface
+{
+    /**
+     * @var DeleteCustomizationFieldHandlerInterface
+     */
+    private $deleteCustomizationFieldHandler;
+
+    /**
+     * @param DeleteCustomizationFieldHandlerInterface $deleteCustomizationFieldHandler
+     */
+    public function __construct(DeleteCustomizationFieldHandlerInterface $deleteCustomizationFieldHandler)
+    {
+        $this->deleteCustomizationFieldHandler = $deleteCustomizationFieldHandler;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(RemoveAllCustomizationFieldsFromProductCommand $command): void
+    {
+        $product = $this->getProduct($command->getProductId());
+
+        $customizationFieldIds = array_map(function ($field) {
+            return (int) $field['id_customization_field'];
+        }, $product->getCustomizationFieldIds());
+
+        foreach ($customizationFieldIds as $customizationFieldId) {
+            $this->deleteCustomizationFieldHandler->handle(new DeleteCustomizationFieldCommand($customizationFieldId));
+        }
+    }
```

Comment:
Are you sure this is the best way?

Reply:
Im sure its the worst :sweat_smile: 
It preserves the object model hooks tho :man_shrugging:

Reply:
@eternoendless Correct. Doing it this way ensures that if a developer extends the `DeleteCustomizationFieldHandler` to enrich the behavior (exemple: decorate it to add a call to an API at the end in order to implement a synchronization between PrestaShop and an ERP) the behavior is kept consistent between the "single delete" and the "bulk delete"

This is **not** the optimized way to do it but I believe optimization should not happen at this level. PrestaShop should rather aim for consistency and robustness.

This however is my opinion 🤔   please tell me if I got it wrong

Reply:
I agree it feels heavy, but it's consistent with the `SetProductCustomizationFieldsHandler` which also uses the removal of customization fields with this handler At least if you need to hook, or override this behaviour it can be done in a single place instead of three

Reply:
Removing one and removing all is not the same action, therefore to me it makes more sense to leave it to the developer to choose to enrich/replace one, the other, or both.

Reply:
need a decision :sun_with_face:

Reply:
I also stand my case regarding ERP sync. Imagine the sync in single delete is performed naively through a synchronous HTTP service call. If you do it this way for bulk delete then you'll have a lot of subsequent HTTP calls instead of a single bulk one. Two use cases, two implementations.